File: service/Lore.Basvuru.Common/DTO/Base/EidDTO.cs
using Lore.Basvuru.Common.Helpers;
using Newtonsoft.Json;

namespace Lore.Basvuru.Common.DTO.Base
{
    /// <summary>
    /// Şifreli ID taşıyıcısı. Frontend ile backend arasında model bazlı çalışır.
    /// - eid set edilince → id otomatik decrypt edilir (request)
    /// - id set edilince  → eid otomatik encrypt edilir (response)
    /// - Sadece eid frontend'e serileştirilir; id backend-only.
    /// </summary>
    public class EidDTO
    {
        private string? _eid;
        private long _id;

        public string? eid
        {
            get => _eid;
            set
            {
                _eid = value;
                _id = string.IsNullOrWhiteSpace(value)
                    ? 0
                    : CryptoHelper.DecryptLong(value);
            }
        }

        [JsonIgnore]
        public long id
        {
            get => _id;
            set
            {
                _id = value;
                _eid = value > 0 ? CryptoHelper.EncryptLong(value) : null;
            }
        }

        // Kolaylık: implicit long dönüşümü → manager'da doğrudan long kullanımı
        public static implicit operator long(EidDTO? dto) => dto?._id ?? 0;

        // Kolaylık: long'dan EidDTO üret
        public static implicit operator EidDTO(long id) => new() { id = id };
    }
}

File: lore-basvuru-ui/src/types/base/EidDTO.ts
/**
 * Şifreli ID taşıyıcısı — backend EidDTO ile birebir eşleşir.
 *
 * Kullanım:
 *  - Request göndermek:  { adimEid: { eid: "abc123" } }
 *  - Response okumak:    response.basvuruEid.eid  → "abc123"
 *
 * Frontend şifre çözme yapmaz; backend her zaman eid string döner.
 */
export interface EidDTO {
  eid?: string | null;
}

/** Kolaylık: string'den EidDTO üret */
export const toEidDTO = (eid: string | null | undefined): EidDTO => ({ eid: eid ?? undefined });

/** Kolaylık: EidDTO'dan eid string çıkar */
export const fromEidDTO = (dto: EidDTO | null | undefined): string => dto?.eid ?? '';
